### ISSUE TITLE
Users have the same rank in the user stats

### ISSUE THREAD

**Issue by Lyncis** [Volentix/venue-server#166]
I've noticed that different users with the same number of tokens have the same overall rank in the user stats. 

To be more detailed,
if I go to http://localhost:8000/api/retrieve/leaderboard-data/
I get the following result:
![image](https://user-images.githubusercontent.com/8875863/43922922-2dea678e-9bee-11e8-9304-f52056b4005b.png)
![image](https://user-images.githubusercontent.com/8875863/43922958-4ac7c586-9bee-11e8-9ff2-a9462bd3b9eb.png)

Then I go to localhost:8000/docs, log in as spiderman, then go to retrieve/stats, I get this:
![image](https://user-images.githubusercontent.com/8875863/43923044-96dc429e-9bee-11e8-8ce6-4fe84b699205.png)

So, in the leaderboard-data spiderman has rank 8, but in the user stats his rank is 9. And it is the same for every user with the same number of tokens as a spiderman.

I don't know if it should be like that or if this needs to be fixed. Leaderboard hilhlighting is not displayed correclty in some cases and it seems like it is because of this rank in user stats.

**Comment by shawnlauzon** [comment]
@joemarct Could you take a look at this?

**Comment by alexdashkov** [comment]
I've tested it on the master and I have the same values in the `leaderboard-data` and `stats`. 
Also, I completely remade `leaderboard-data` in the `performance` branch, so I prefer to wait for the merge and after that develop the changes if required. 

And a question, @shawnlauzon is it correct that users with the same points have a different rank?

**Comment by shawnlauzon** [comment]
Ok, that seems reasonable. @alexdashkov please create a PR for merging into master and ask @joemarct to review it.

Yes, users with same points will have different ranks; the ranking can be arbitrary (although ideally the first user who earned a certain number of points should be ranked higher), but consistent.

**Comment by alexdashkov** [comment]
@shawnlauzon do you mean a PR with the leaderboard data? @realrhys has said that that he will merge a `performance` branch to the master by the end of testing.

**Comment by shawnlauzon** [comment]
Yes, with the leaderboard data. Now this is more than a performance fix, it's possibly a correctness fix. So I want it in master so that we can complete our integration testing. Performance testing can happen in parallel in the `performance` branch, and if there are additional performance fixes, they can be merged later.

**Comment by alexdashkov** [comment]
@shawnlauzon I'm not sure yet that this a bug fix as I can't reproduce the problem. I'm waiting for the @Lyncis's response. Does it make sense?

**Comment by Lyncis** [comment]
I pulled the last changes on the server, still have the same issue,

Trying to explain it again.

I go to http://localhost:8000/docs/
Authenticate as ironman: 
![image](https://user-images.githubusercontent.com/8875863/44044629-2c3565de-9ef4-11e8-976f-77759c9e903a.png)

Then I retrieve leaderboard-data with this user, I see that his rank is 10.
![image](https://user-images.githubusercontent.com/8875863/44044681-6028f216-9ef4-11e8-9fc9-2295664ad465.png)

Then I go to retrieve user stats and see that his overall_rank is 10

![image](https://user-images.githubusercontent.com/8875863/44044780-a0af2c1a-9ef4-11e8-9f8c-29a8c2a4521e.png)

Then if I log in as spiderman and retrieve leaderboard-data, his rank is 9 here
![image](https://user-images.githubusercontent.com/8875863/44044902-fa887232-9ef4-11e8-8fe3-19b5b89c7a94.png)

But if I go to the user stats, his overall_rank is 10 as it was for ironman
![image](https://user-images.githubusercontent.com/8875863/44044943-1d47fc52-9ef5-11e8-85f3-c7f1c3596ed4.png)

Same is for hawkeye and other who has the same number of tokens. As for other users, their rank is displayed correctly.

I also noticed that when I am retrieving leadreboard for these users, they quite often have different ranks. For example, spiderman can have rank 9, and on the next time he can have rank 7. In the user_stats though, it is always the same for all of them.

**Comment by Lyncis** [comment]
Seems like this is an issue only locally, everything works fine on uat and dev.
